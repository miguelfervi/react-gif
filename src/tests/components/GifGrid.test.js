import React from "react";

import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en el GifGrid", () => {
  const category = "One Punch";

  test("should correctly", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category}></GifGrid>);
    expect(wrapper).toMatchSnapshot();
  });

  test("should correctly with images", () => {
    
    const gifs = [{
        id: 'ABC',
        url: 'https://github.com/',
        title: 'cualquier cosa'
    }]

    useFetchGifs.mockReturnValue({
        data: gifs,
        loading: false,
      });
    
    const wrapper = shallow(<GifGrid category={category}></GifGrid>);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
