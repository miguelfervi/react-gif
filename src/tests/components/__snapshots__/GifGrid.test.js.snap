// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifGrid should correctly 1`] = `
<div>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Cargando... 
  </p>
  <div
    className="card-grid"
  />
</div>
`;
